community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1917,3 +1917,52 @@ def update_config(cfg: NSDict) -> None:
 def scontrol_reconfigure(lkp: Lookup) -> None:
     log.info("Running scontrol reconfigure")
     run(f"{lkp.scontrol} reconfigure", timeout=30)
+
+
+class NodeState(str, enum.Enum):
+    """
+    Representation of Slurm NODE_STATE
+
+    For source of truth with 1-line explanation, see:
+    https://github.com/SchedMD/slurm/blob/master/slurm/slurm.h#L980
+    For string representation to match, see:
+    https://github.com/SchedMD/slurm/blob/master/src/plugins/data_parser/v0.0.42/parsers.c#L7449
+
+    NOTE: Copy of StrEnum implementation from python 3.11 stripped of extra features.
+    Additionally support "unknown" items by defining `_missing_`.
+    See tests for contract.
+    """
+    def __new__(cls, value):
+        return str.__new__(cls, value)
+    
+    @classmethod
+    def _missing_(cls, value):
+        o = str.__new__(cls, value)
+        o._name_ = value
+        return o
+
+    def _generate_next_value_(name, *_):
+        return name
+    
+    def __repr__(self):
+        return self.__str__()
+    
+    # *** Base states
```

Comment:
Will other node states like unknown or end be added or are they not needed? (from https://github.com/SchedMD/slurm/blob/master/slurm/slurm.h#L980)